### PR TITLE
Feature/decommission cache

### PR DESCRIPTION
**Pull request by AWitcherONS**
### What

The removal of Enable Legacy Cache  now that legacy cache is permanently active

### How to review
There is no new code just the removal of existing code 
build for both cms and reader should be successful
Unit test for cms/reader should be successful

5. Manual testing 
      Zebedee Read no action but should work as expected i.e. web 
      Zebedee CMS :-
                Publication Notification has been amended 
                       Manual Publication of simple collection
                       Scheduled Publication of simple collection

### Who can review

any one but me

### CODE REVIEW

File: zebedee-cms/src/main/java/com/github/onsdigital/zebedee/model/publishing/PublishNotification.java [hunk 1]
```diff
@@ -38,9 +38,9 @@ public class PublishNotification {
     }
 
     public PublishNotification(Collection collection, List<String> urisToUpdate, List<ContentDetail> urisToDelete) {
-        if (Configuration.isLegacyCacheAPIEnabled()) {
+//        if (Configuration.isLegacyCacheAPIEnabled()) {
             this.legacyCacheApiPayloads = new LegacyCacheApiPayloadBuilder.Builder().collection(collection).build().getPayloads();
-        }
+//        }
```
Comment (lindenmckenzie) [inline zebedee-cms/src/main/java/com/github/onsdigital/zebedee/model/publishing/PublishNotification.java:43]: Should be removing this, not just commenting it out.

File: zebedee-cms/src/main/java/com/github/onsdigital/zebedee/model/publishing/PublishNotification.java [hunk 2]
```diff
@@ -55,7 +55,7 @@ public PublishNotification(Collection collection) {
     }
 
     public void sendNotification(EventType eventType) {
-        if (Configuration.isLegacyCacheAPIEnabled()) {
+//        if (Configuration.isLegacyCacheAPIEnabled()) {
```
Comment (lindenmckenzie) [inline zebedee-cms/src/main/java/com/github/onsdigital/zebedee/model/publishing/PublishNotification.java:58]: Should be removing this, not just commenting it out.

**Review by lindenmckenzie — CHANGES_REQUESTED** [review]
Some comments on comments :D - also don't forget the readme.

File: zebedee-cms/src/main/java/com/github/onsdigital/zebedee/model/publishing/PublishNotification.java [hunk 3]
```diff
@@ -55,7 +52,6 @@ public PublishNotification(Collection collection) {
     }
 
     public void sendNotification(EventType eventType) {
```
Comment (lindenmckenzie) [inline zebedee-cms/src/main/java/com/github/onsdigital/zebedee/model/publishing/PublishNotification.java:54]: This needs a reformat now the if condition has been removed.

**Review by lindenmckenzie — CHANGES_REQUESTED** [review]
Reformat required - still missing the readme as well

File: zebedee-cms/src/main/java/com/github/onsdigital/zebedee/model/publishing/PublishNotification.java [hunk 4]
```diff
@@ -38,13 +38,8 @@ public class PublishNotification {
     }
 
     public PublishNotification(Collection collection, List<String> urisToUpdate, List<ContentDetail> urisToDelete) {
-        if (Configuration.isLegacyCacheAPIEnabled()) {
-            this.legacyCacheApiPayloads = new LegacyCacheApiPayloadBuilder.Builder().collection(collection).build().getPayloads();
-        }
-
-        // Babbage still reindexes on a published event.
-        // TODO: remove this when it doesn't anymore.
-        // Delay the clearing of the cache after publish to minimise load on the server while publishing.
```
Comment (lindenmckenzie) [inline zebedee-cms/src/main/java/com/github/onsdigital/zebedee/model/publishing/PublishNotification.java:47]: I think this comment is still relevant.

File: zebedee-cms/src/main/java/com/github/onsdigital/zebedee/model/publishing/PublishNotification.java [hunk 5]
```diff
@@ -55,18 +50,12 @@ public PublishNotification(Collection collection) {
...
     public void sendNotification(EventType eventType) {
-        if (Configuration.isLegacyCacheAPIEnabled()) {
             sendRequestToLegacyCacheApi(eventType);
-            if (eventType.equals(EventType.APPROVED) || eventType.equals(EventType.PUBLISHED)) {
+            if (eventType.equals(EventType.APPROVED) ) {
                 // Babbage still relies on
-                //   1. 'approved' for storing list of (upcoming) pages
-                //   2. 'published' event for reindexing pages from 1.
-                // TODO: remove this when it doesn't anymore.
+                //  'approved' for storing list of (upcoming) pages
```
Comment (lindenmckenzie) [inline zebedee-cms/src/main/java/com/github/onsdigital/zebedee/model/publishing/PublishNotification.java:56]: I think these comments are still relevant.

File: zebedee-cms/src/main/java/com/github/onsdigital/zebedee/model/publishing/PublishNotification.java [hunk 6]
```diff
@@ -55,18 +50,12 @@ public PublishNotification(Collection collection) {
     }
 
     public void sendNotification(EventType eventType) {
-        if (Configuration.isLegacyCacheAPIEnabled()) {
             sendRequestToLegacyCacheApi(eventType);
-            if (eventType.equals(EventType.APPROVED) || eventType.equals(EventType.PUBLISHED)) {
```
Comment (lindenmckenzie) [inline zebedee-cms/src/main/java/com/github/onsdigital/zebedee/model/publishing/PublishNotification.java:60]: We still need to send the message on PUBLISHED.

**Review by lindenmckenzie — CHANGES_REQUESTED** [review]
Few comments - happy to take these forward in the team now or for you to take them forward, up to you. Additionally - the readme hasn't been updated.

**Review by Oluwawunmi — DISMISSED** [review]

**Review by Oluwawunmi — APPROVED** [review]

**Review by lindenmckenzie — APPROVED** [review]

**Pull request merged**